homeassistant/components/switch/acer_beamer.py

```diff
@@ -0,0 +1,145 @@
+"""
+Use serial protocol of acer beamer to obtain state of the beamer.
+
+This component allows to control almost all beamers from acer using
+their RS232 serial communication protocoll.
+"""
+
+import logging
+import re
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import (STATE_ON, STATE_OFF, STATE_UNKNOWN,
+                                 CONF_NAME, CONF_FILENAME)
+
+LAMP_HOURS = 'Lamp Hours'
+INPUT_SOURCE = 'Input Source'
+ECO_MODE = 'ECO Mode'
+MODEL = 'Model'
+LAMP = 'Lamp'
+
+# Commands known to the beamer
+CMD_DICT = {LAMP: '* 0 Lamp ?\r',
+            LAMP_HOURS: '* 0 Lamp\r',
+            INPUT_SOURCE: '* 0 Src ?\r',
+            ECO_MODE: '* 0 IR 052\r',
+            MODEL: '* 0 IR 035\r',
+            STATE_ON: '* 0 IR 001\r',
+            STATE_OFF: '* 0 IR 002\r'}
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['pyserial']
```

Comment:
PySerial 3 introduced a new API. Are you sure that your component works if you are using PySerial 2 or do you need to restrict the version ?